trie/leafNode.go

```diff
@@ -201,23 +173,10 @@ func (ln *leafNode) getEncodedNode() ([]byte, error) {
 	return marshaledNode, nil
 }
 
-func (ln *leafNode) resolveCollapsed(_ byte, _ common.TrieStorageInteractor) error {
-	return nil
-}
-
-func (ln *leafNode) isCollapsed() bool {
-	return false
-}
-
-func (ln *leafNode) isPosCollapsed(_ int) bool {
-	return false
-}
-
 func (ln *leafNode) tryGet(key []byte, currentDepth uint32, _ common.TrieStorageInteractor) (value []byte, maxDepth uint32, err error) {
-	err = ln.isEmptyOrNil()
-	if err != nil {
-		return nil, currentDepth, fmt.Errorf("tryGet error %w", err)
-	}
+	ln.mutex.RLock()
+	defer ln.mutex.RUnlock()
```

Comment:
do we need mutex here?

Reply:
this mutex is protecting `hash` and `dirty` fields, right?

Reply:
It protects everything that can change on the node, so `ln.Key` & `ln.Value` also